lib/TaskProcessing/SlideDeckGenerationProvider.php

```diff
@@ -80,10 +80,15 @@ public function process(?string $userId, array $input, callable $reportProgress)
 			throw new \RuntimeException('Invalid input, expected "text" key with string value');
 		}
 
-		$response = $this->slideDeckService->generateSlideDeck(
-			$userId,
-			$input['text'],
-		);
+		try {
+			$response = $this->slideDeckService->generateSlideDeck(
+				$userId,
+				$input['text'],
+			);
+		} catch (\RuntimeException $e) {
+			// We can retry in case of faulty LLM JSON
+			$this->process($userId, $input, $reportProgress);
```

Comment:
```suggestion
			return $this->process($userId, $input, $reportProgress);
```
But you risk an infinite loop here.